plugins/reports/counter/classes/releases/CounterRelease.inc.php

```diff
@@ -0,0 +1,254 @@
+	 */
+	function getVendorLogoUrl() {
+		return $this->_getVendorComponent('logo');
+	}
+
+	/**
+	 * Get the COUNTER metric type for an Statistics file type
+	 * @param $filetype string
+	 * @return string
+	 */
+	function getKeyForFiletype($filetype) {
+		switch ($filetype) {
+			case STATISTICS_FILE_TYPE_HTML:
+				$metricTypeKey = 'ft_html';
+				break;
+			case STATISTICS_FILE_TYPE_PDF:
+				$metricTypeKey = 'ft_pdf';
+				break;
+			case STATISTICS_FILE_TYPE_OTHER:
+			default:
+				$metricTypeKey = 'other';
+		}
+		return $metricTypeKey;
+	}
+	
+	/**
+	 * Get the Vendor Componet by key
+	 * @param $key string
+	 * @return mixed
+	 */
+	protected function _getVendorComponent($key) {
+		$request = PKPApplication::getRequest();
+		$site = $request->getSite();
+		switch ($key) {
+			case 'name':
+				return $site->getLocalizedTitle();
+			case 'id':
+				return $request->getBaseUrl();
+			case 'contacts':
+				try {
+					$contact = new COUNTER\Contact($site->getLocalizedContactName(), $site->getLocalizedContactEmail());
+				} catch (Exception $e) {
+					$this->setError($e);
+					$contact = array();
+				}
+				return $contact;
+			case 'website':
+				return $request->getBaseUrl();
+			case 'logo':
+				return '';
+			default:
+				return;
+		}
+	}
+
+	/**
+	 * Abstract method must be implemented in the child class
+	 * Get the report title
+	 * @return $string
+	 */
+	function getTitle() {
+		assert(false);
+	}
+
+	/**
+	 * Abstract method must be implemented by subclass
+	 * @return string COUNTER XML
+	 */
+	function getMetricsXML($columns = array(), $filters = array(), $orderBy = array(), $range = null) {
+		assert(false);
+	}
+
+	/**
+	 * Get an array of errors
+	 * @return array of Exceptions
+	 */
+	function getErrors() {
+		return $this->_errors ? $this->_errors : array();
+	}
+
+	/**
+	 * Set an errors condition; Proper Exception handling is deferred until the OJS 3.0 Release
+	 * @param $error Exception
+	 */
+	function setError($error) {
+		if (!$this->_errors) {
+			$this->_errors = array();
+		}
+		array_push($this->_errors, $error);
+	}
+
+	/**
+	 * Ensure that the $filters do not exceed the current Context
+	 * @param array() $filters
+	 * @return array()
+	 */
+	protected function filterForContext($filters) {
+		$request = PKPApplication::getRequest();
+		$journal = $request->getJournal();
+		$journalId = $journal ? $journal->getJournalId() : '';
+		// If the request context is at the journal level, the dimension context id must be that same journal id
+		if ($journalId) {
+			if (isset($filters[STATISTICS_DIMENSION_CONTEXT_ID]) && $filters[STATISTICS_DIMENSION_CONTEXT_ID] != $journalId) {
+				$this->setError(new Exception(__('plugins.reports.counter.generic.exception.filter'), COUNTER_EXCEPTION_WARNING | COUNTER_EXCEPTION_BAD_FILTERS));
+			}
+			$filters[STATISTICS_DIMENSION_CONTEXT_ID] = $journalId;
+			if (isset($filters[STATISTICS_DIMENSION_ASSOC_ID]) && $filters[STATISTICS_DIMENSION_ASSOC_ID] != ASSOC_TYPE_GALLEY) {
```

Comment:
Hmmm... My memory is fuzzy on this one.  I think my concern was ensuring that a COUNTER report would always be concerned with certain types of downloads. Not only did I get the code wrong, I think the scope is wrong.  It is the individual reports that should define what types of ASSOC_IDs they care about.  The abstract report shouldn't care.  I'll remove this check.